frontend-maven-plugin/src/main/java/com/github/eirslett/maven/plugins/frontend/mojo/NpmMojo.java

```diff
@@ -55,14 +58,22 @@ protected boolean skipExecution() {
     }
 
     @Override
-    public synchronized void execute(FrontendPluginFactory factory) throws TaskRunnerException {
+    public synchronized void execute(FrontendPluginFactory factory) throws Exception {
         File packageJson = new File(workingDirectory, "package.json");
-        if (buildContext == null || buildContext.hasDelta(packageJson) || !buildContext.isIncremental()) {
+
+        boolean incrementalEnabled = incrementalBuildEnabled(buildContext);
+        boolean willBeIncremental = incrementalEnabled && buildContext.hasDelta(packageJson);
```

Comment:
Should we also check delta of lock files? If versions are not locked lock files might change and require different package versions without any changes made in package.json